app/resources/freetextsearch.py

```diff
@@ -76,11 +81,11 @@ def post(self):
         start_time = time.time()
         kwargs = request.get_json(force=True)
         
-        kwargs['fields'] = ['id', 'approved_symbol']; #do not want any other fields
-        
-        filter = ['all']
-        if ('filter' in kwargs):
-            filter = kwargs['filter']
+        filter = [kwargs.pop('filter')] or ['all']
+        if 'target' in filter:
+            kwargs['fields'] = ['id','approved_symbol']; #do not want any other fields
+        elif 'disease' in filter:
+            kwargs['fields'] = ['id','name'];
```

Comment:
_id_ is always there, I think you can simplify with:

```
kwargs['fields'] = ['approved_symbol', 'name'];
```
that should work with targets and diseases (even if diseases don't have _approved_symbol_, it just gets ignored)